provider/dynamodb_test.go

```diff
@@ -0,0 +1,115 @@
+package provider
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbiface"
+	"github.com/containous/traefik/types"
+)
+
+type mockDynamoDBCLient struct {
+	dynamodbiface.DynamoDBAPI
+}
+
+var backend = &types.Backend{
+	HealthCheck: &types.HealthCheck{
+		URL: "/build",
+	},
+	Servers: map[string]types.Server{
+		"server1": {
+			URL: "http://test.traefik.io",
+		},
+	},
+}
+
+var frontend = &types.Frontend{
+	EntryPoints: []string{"http"},
+	Backend:     "test.traefik.io",
+	Routes: map[string]types.Route{
+		"route1": {
+			Rule: "Host:test.traefik.io",
+		},
+	},
+}
+
+var testWithError = false
```

Comment:
A problem with the approach of using a global variable is that it can make efforts to parallelize tests very difficult. It also complicates the current approach because `TestLoadDynamoConfig` now combines both states (`testWithError` equals true and false, respectively) into a single test, which makes it impossible to run the tests individually or see the outcome of the second before the first test.

Since we already have a testing struct, could you please move the variable into `mockDynamoDBCLient` and then set the value accordingly in two separate tests (e.g., `TestLoadDynamoConfigSuccessful` and `TestLoadDynamoConfigFailure`)?